### ISSUE TITLE
Simplify mechanism of providing params and retrieving result

### ISSUE THREAD

**Issue by kwarunek** [DreamLab/AsyncOpenStackClient#12]
Maybe create some sort of Resource (of python-simple-rest-client) that will handle passing arguments and fetching data.

Current usage:

```
 servers = await self.nova.api.servers.list(params={'name': '^my'})
 for server in servers.body["servers"]:
     print(server)
```

Expected usage:    
```
 servers = await self.nova.api.servers.list(name='^my')
 for server in servers:
     print(server)
```

**Comment by evemorgen** [comment]
This issue shall be resolved by this PR: #20

**Comment by evemorgen** [comment]
By virtue of #20  I'm closing this issue as resolved :)
